Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,42 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    if File.exist? "#{var}/postgres/PG_VERSION"
+      pg_version = (var/"postgres/PG_VERSION").read.chomp
+    end
+
+    if (var/"postgres").exist? && (var/"postgres-#{pg_version}").exist?
+      puts "We have not attempted to migrate your data due to your multi-postgresql setup."
+    end
+
+    # With the 9.5 release, we'll need to adapt logic to make sure we preserve the
+    # 9.4 var directories, without breaking postgres. This should be able to be done
+    # by modifying some lines below to var/"postgres-4.9" instead of var/"postgres"
+    old_path = var/"postgres"
+    new_path = var/"postgres-#{version_suffix}"
+    # This should always be the last full version shipped by Brew, i.e. 9.4.0.
+    prior_ver = "9.4.0"
```

Comment:
Indeed. This was largely intended to be manual, to be honest. It's complicated by the fact ` PG_VERSION ` isn't guaranteed to exist, and when it does, ` PG_VERSION ` only contains the major version, x.x rather than x.x.x, so it isn't a super super easy thing to consistently automate. I guess we could implement a check for the other version in ` #{HOMEBREW_CELLAR}/postgresql ` at the time of install, but I don't know how you feel about the amount of logic we're burning through in ` postinstall ` at this point.